File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-2.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        n = len(arr)
        nums = [0] * n
        rightMax = -1
        for i in reversed(range(n)):
            nums[i] = rightMax
            rightMax = max(rightMax, arr[i])
        return nums

File: Data Structures & Algorithms/single-element-in-a-sorted-array/submission-2.rs
impl Solution {
    pub fn single_non_duplicate(nums: Vec<i32>) -> i32 {
        let mut l = 0;
        let mut r = nums.len() - 1;
        while l < r {
            let mut m = l + (r - l) / 2;
            if m % 2 != 0 {  
                m = m - 1;
            }
            if nums[m] == nums[m + 1] {
                l = m + 2;
            } else {
                r = m;
            }
        }
        return nums[l];
    }
}
